flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/tiered/file/ProducerMergedPartitionFileReader.java

```diff
@@ -60,9 +141,122 @@ public long getPriority(
             TieredStorageSubpartitionId subpartitionId,
             int segmentId,
             int bufferIndex) {
-        return 0;
+        Tuple2<TieredStorageSubpartitionId, Integer> cacheKey =
+                Tuple2.of(subpartitionId, bufferIndex);
+        RegionCache regionCache = regionCaches.get(cacheKey);
```

Comment:
It's a bit implicit how is this different from `tryGetRegionCache`

Reply:
I'm not sure about not fetching the index when the offset is not cached. In cases where the region contains only a few buffers, this may completely invalid the io scheduling.

Reply:
Fixed.